### PR TITLE
Added zero.symint and modified aten::trapz to use symbolic ints

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82389
* #82209
* __->__ #82054

### CODE REVIEW

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -741,8 +741,8 @@ def f(a, b):
     xfail('tensordot', ''),  # aten.size.default - couldn't find symbolic meta function/decomposition
     xfail('tile', ''),  # aten.repeat.default - couldn't find symbolic meta function/decomposition
     xfail('topk', ''),  # aten.topk.default - couldn't find symbolic meta function/decomposition
-    xfail('trapezoid', ''),  # aten.size.default - couldn't find symbolic meta function/decomposition
     xfail('trapz', ''),  # aten.size.default - couldn't find symbolic meta function/decomposition
+    xfail('trapezoid', ''),  # aten.size.default - couldn't find symbolic meta function/decomposition
```
Comment (ezyang) [inline test/test_proxy_tensor.py:745]: nani?

File: aten/src/ATen/native/TensorFactories.cpp [hunk 2]
```diff
@@ -1090,6 +1090,18 @@ Tensor zeros(IntArrayRef size,
   return result.zero_();
 }
 
+// TODO: Seems pretty bad that we have to have this separate function.
```
Comment (ezyang) [inline aten/src/ATen/native/TensorFactories.cpp:1093]: How come you can't have old zeros call this?

**Review by ezyang — APPROVED** [review]

File: test/test_proxy_tensor.py (hunk 1 above)
Comment (Chillee) [reply test/test_proxy_tensor.py:745]: uh... probably a rebase issue lol.

File: aten/src/ATen/native/TensorFactories.cpp (hunk 2 above)
Comment (Chillee) [reply aten/src/ATen/native/TensorFactories.cpp:1093]: Good point, will do.

**Pull request closed without merging**